hello-world/2023-02-20_testable_clock.md

```diff
@@ -0,0 +1,130 @@
+﻿# A testable clock
+Every developer that ever build an application that has any time dependency
+knows that testing such dependencies is not straight forward. The most pressing
+issue is arguably that most programming languages (including C#, PHP, Java, SQL)
+do not allow you to change the behaviour of the (system) clock/date time provider.
+So how to overcome this problem?
```

Comment:
We can overcome this problem by creating an abstraction for clock.
I'll show you several ways, including my favorite one, the Adjustable singleton delegate.

1. Create a stubable date time provider.
2. Create an adjustable global, static clock function.
3. Create an adjustable singleton delegate.

To make sure all occurences of DateTime.Now (and DateTime.Today) are refactored, we can enable sonarqube rule S6354.

.. And then

Reply:
That's a great suggestion. I applied it.